docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,297 @@
+     an API call, but costs real money.
+   * A bare-metal disk device for which provisioning means sending a ticket to
+     your sys-admin, who eventually types some commands into a control
+     application and responds to your ticket with a device ID.
+
+## Status
+
+This document is somewhat aspirational.  The state of the system at the time of
+this writing is NOT fully aligned with this model, but we intend for the system
+to trend towards it.
+
+## General model
+
+Both storage and networking resources can be handled with a common model.
+
+```
+                       +----------+
+        +--create----> | resource |
+        |              +----------+
+   +----+----+               ^
+   | manager |---bind------> :
+   +----+----+               v
+        |              +----------+
+        +---watch----> |  claim   | <--+
+                       +----------+    |
+                             ^         |
+                             |         +---- user-created
+                             |         |
+                       +----------+    |
+                       | consumer | <--+
+                       +----------+
+```
+
+In this model a `consumer` object (created by a user in the API) files a
+`claim` (also created by the user) for some resource.  Claims are abstract by
+nature (see the storage and networking forms of this diagram below) and
+describe what kind of resource is needed, but (generally) not the identity of
+the specific resource instance.  A `Manager` entity (a process, not part of the
+API) is responsible for satisfying the claim by "binding" the claim to a
+specific resource.
+
+You can sort of see this same pattern in the scheduling of `Pod`s onto `Node`s.
+A `Pod` is a claim against resources on a `Node`, and is bound by the scheduler.
+
+## Storage
+
+Storage in Kubernetes will be represented by `PersistentVolume`s.  A user can
+create a `PersistentVolumeClaim` for storage and use that claim from a `Pod`.
+A `VolumeManager` is responsible for binding `PersistentVolumeClaim`s to
+`PersistentVolume`s.  If the `PersistentVolumeClaim` can not be satisified by
+any existing `PersistentVolume`, the `VolumeManager` may provision a new one.
+
+```
+                       +------------------+
+        +--create----> | PersistentVolume |
+        |              +------------------+
+   +----+----+               ^
+   | VolMgr  |---bind------> :
+   +----+----+               v
+        |              +------------------------+
+        +---watch----> | PersistentVolumeClaim  | <--+
+                       +------------------------+    |
+                                 ^                   |
+                                 |         +---------+-- user-created
+                                 |         |
+                           +----------+    |
+                           |    Pod   | <--+
+                           +----------+
+```
+
+### PersistentVolumeClaim
```

Comment:
I think someone new to this might ask why a PersistentVolumeClaim is a separate object rather than being inlined into the Pod description. You might want to address that.